core/vm/evm.go

```diff
@@ -227,10 +227,12 @@ func (self *EVM) Main(trx *Transaction) (ret ExecutionResult) {
 	gas_left -= gas_intrinsic
 
 	if contract_creation {
+		// setting nonce to current trx nonce to generate correct address for new contract. Nonce incremented inside of `create` later
+		caller.SetNonce(self.trx.Nonce)
 		ret.CodeRetval, ret.NewContractAddr, gas_left, err = self.create_1(caller, self.trx.Input, gas_left, self.trx.Value)
```

Comment:
last time we were talking about it you said it is better to use tx_nonce for contract address calculation, but you did not do this change ?

Reply:
This is done by line above this one. We are setting account nonce to actual transaction nonce and incrementing it later  in `create()`

Reply:
aah I see now. Hmm isnt it better/cleaner to rather delete that nonce Increment from create and use this:
`caller.SetNonce(bigutil.Add(self.trx.Nonce, big.NewInt(1)))`

for both contract creation and standard tx. We shouldnt probably use nonce increment at all as we allow nonce skipping - it is kind of misleading